bindings/web5_wasm/src/crypto/key_managers.rs

```diff
@@ -0,0 +1,75 @@
+use super::{dsa::WasmSigner, jwk::WasmJwk};
+use crate::errors::{map_err, Result};
+use std::sync::Arc;
+use wasm_bindgen::prelude::wasm_bindgen;
+use web5::crypto::{
+    jwk::Jwk,
+    key_managers::{in_memory_key_manager::InMemoryKeyManager, KeyManager},
+};
+
+#[wasm_bindgen]
+pub struct WasmKeyManager {
+    inner: Arc<dyn KeyManager>,
+}
+
+impl From<InMemoryKeyManager> for WasmKeyManager {
+    fn from(value: InMemoryKeyManager) -> Self {
+        Self {
+            inner: Arc::new(value),
+        }
+    }
+}
+
+#[wasm_bindgen]
+impl WasmKeyManager {
+    pub fn import_private_jwk(&self, private_jwk: WasmJwk) -> Result<WasmJwk> {
+        Ok(self
+            .inner
+            .import_private_jwk(private_jwk.into())
+            .map_err(map_err)?
+            .into())
+    }
+
+    pub fn get_signer(&self, public_jwk: WasmJwk) -> Result<WasmSigner> {
+        Ok(self
+            .inner
+            .get_signer(public_jwk.into())
+            .map_err(map_err)?
+            .into())
+    }
+}
+
+#[wasm_bindgen]
+pub fn new_in_memory_key_manager() -> Result<WasmKeyManager> {
+    Ok(InMemoryKeyManager::new().into())
+}
+
+#[wasm_bindgen]
+extern "C" {
```

Comment:
here's an example of a _foreign_ implementation, for which we'll use this same pattern for a foreign `fetch` implementation